### PR TITLE
OpenTelemetry support

### PR DESCRIPTION
**Pull request by samuelrince**
This PR adds support for OpenTelemetry into EcoLogits. This will allow users to export EcoLogits metrics directly into their monitoring stack.

Thanks to [Afoodi](https://afoodi.io/) for contributing to this new feature! 🤗 

### Documentation summary

**Initialize OpenTelemetry within EcoLogits:**

```python
from ecologits import EcoLogits
from openai import OpenAI

# Configure OpenTelemetry endpoint in EcoLogits
EcoLogits.init(
    providers=["openai"],
    opentelemetry_endpoint='https://localhost:4318/v1/metrics'
)

client = OpenAI()

# All requests will log ecologits metrics
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[{"role": "user", "content": "Tell me a funny joke!"}]
)
```

**Tracked metrics:**

| Metric name                         | Type    | Description                                         |
|-------------------------------------|---------|-----------------------------------------------------|
| `ecologits_requests_total`          | Counter | Number of requests                                  |
| `ecologits_input_tokens_ratio`      | Gauge   | Number of input tokens                              |
| `ecologits_output_tokens_ratio`     | Gauge   | Number of output tokens                             |
| `ecologits_request_latency_seconds` | Gauge   | Request latency in seconds                          |
| `ecologits_energy_joules`           | Gauge   | Average energy consumption of the request in joules |
| `ecologits_gwp_gCO2eq`              | Gauge   | Average GWP impacts of the request in gCO2eq        |
| `ecologits_adpe_gSbeq`              | Gauge   | Average ADPe impacts of the request in gSbeq        |
| `ecologits_pe_joules`               | Gauge   | Average PE impacts of the request in joules         |

**Use OpenTelemetry labels:**

```python
from ecologits import EcoLogits
from openai import OpenAI

EcoLogits.init(
    providers=["openai"], 
    opentelemetry_endpoint="http://localhost:4318/v1/metrics"
)
client = OpenAI()

# with a context manager
with EcoLogits.label(my_custom_label="test"):
    response = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[{"role": "user", "content": "Tell me a funny joke"}]
    )
    
# or, with a decorator
@EcoLogits.label(task="summarization")
def summarize(text: str) -> str:
    response = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[{"role": "user", "content": f"Summarize this:\n\n{text}"}]
    )
    return response.choices[0].message.content
```

### Work in progress:

- [x] Make OpenTelemetry an optional dependency 
- [x] Integration with all other providers

### CODE REVIEW

**Comment by samuelrince** [comment]
Ping @jaydesl and @sauraisg, can you please review the changes? :) 

We still have a bit of work on our side to adapt this to all providers, but it will be done and released by the end of next week for sure!

**Comment by samuelrince** [comment]
Hey @adrienbanse, if you could double-check this PR (and especially the documentation), that would be great! For "integration tests," I have everything working locally, and I will make a special private repo for now.

**Comment by henrikrexed** [comment]
I would recommend to not report a counter , but rather a Gauge or Histograms.
Counter are create for reporting the per second value

File: docs/tutorial/opentelemetry.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+client = OpenAI()
+
+with EcoLogits.label(my_custom_label="test"): # (1)! 
+    response = client.chat.completions.create(
+        model="gpt-4o-mini",
+        messages=[{"role": "user", "content": "Tell me a funny joke"}]
+    )
+```
+
+1. Also work with asynchronous code with `async with` syntax.
```
Comment (adrienbanse) [inline docs/tutorial/opentelemetry.md:94]: Maybe works instead of work (+ same comment for below)

File: docs/tutorial/opentelemetry.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# OpenTelemetry support
+
+With the support of [OpenTelemetry :octicons-link-external-16:](https://opentelemetry.io/) in EcoLogits you can monitor request metrics such as the number of input/output tokens or the environmental impacts in time.
```
[the remark this answers is not in the document]
Comment (samuelrince) [reply docs/tutorial/opentelemetry.md:3]: Yes you are right! I'll make an issue for a future release.
